addon/components/frost-list-pagination.js

```diff
@@ -77,14 +81,50 @@ export default Component.extend({
   @computed('_offset', '_end', 'total')
   _paginationText (_offset, _end, total) {
     return total === 0 ? '0 results found' : `${_offset} to ${_end} of ${total}`
-  }
-
+  },
+  @readOnly
+  @computed('total', 'itemsPerPage')
+  _lastPage (total, itemsPerPage) {
+    return Math.floor((total - 1) / itemsPerPage)
+  },
   // == Functions =============================================================
-
+  _sendDebounceOnChange (page) {
+    this.sendAction('onChange', page)
+    this.set('pageClickCounter', 0)
+  },
+  /* eslint-disable complexity */
+  _handleDebouncedOnChange (page, debounceInterval) {
+    const currentPage = this.get('page')
+
+    if (Math.abs(currentPage - page) > 1) {
+      const debouncedCall = this.get('_debouncedPaging')
+      run.cancel(debouncedCall)
+      this._sendDebounceOnChange(page)
+    } else {
+      const totalPages = this.get('_lastPage')
+      let counter = this.get('pageClickCounter') || 0
+      const isGoingBack = currentPage > page
+      isGoingBack ? counter-- : counter++
+      this.set('pageClickCounter', counter)
+      let newPage = currentPage + counter
+      newPage = newPage < 0 ? 0 : newPage
+      newPage = newPage > totalPages ? totalPages : newPage
+      this.set('_debouncedPaging', run.debounce(this, this._sendDebounceOnChange, newPage, debounceInterval))
```

Comment:
Multiple `this.set()` calls should be combined into a single `this.setProperties()` call.